### ISSUE TITLE
Android: TabView (under load) crashes

### ISSUE THREAD

**Issue by rborn** [NativeScript/NativeScript#2085]
### Which platform(s) does your issue occur on?
Android

### Please provide the following version numbers that your issue occurs with:
- CLI: 2.0.0
- Cross-platform modules: 2.0.0
- Runtime(s): tns-android: 2.0.0
- Plugin(s): "nativescript-google-maps-sdk": "^1.2.8" (**only to add heavy load to the app**)

### Please tell us how to recreate the issue in as much detail as possible.

Hi guys,

It seems that if you try to use a ListView inside a TabView, once the app starts to have some load it will start to behave weird.

**Steps to reproduce:**
- create a 4 tabs app
- put in the first tab a map (be sure you have a real gmap key, so you have the tiles loading to put some load on the app)
- put in the 4th tab a ListView with a custom itemTemplate with 2 labels and an image
- set the model to this: 
~~~
viewModel.listItems = [{
		title: '1',
		description: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
		image: 'http://lorempixel.com/200/100/'
	},
...
~~~
- launch the app and switch between the 1st tab and the 4th several times

**Result (one of the below or both):**
- The app won't fully complete the animation of the tabs 
- The app will crash with the attached error

 
Let me know if there is anything I can help with 😄 
Thank you.

~~~
05-09 19:38:44.739: W/dalvikvm(7188): threadid=1: thread exiting with uncaught exception (group=0x4167edb8)
05-09 19:38:44.749: W/System.err(7188): com.tns.NativeScriptException: 
05-09 19:38:44.749: W/System.err(7188): Calling js method instantiateItem failed
05-09 19:38:44.749: W/System.err(7188): Error: java.lang.ClassCastException: android.view.AbsSavedState$1 cannot be cast to android.widget.AbsListView$SavedState
05-09 19:38:44.749: W/System.err(7188):     android.widget.AbsListView.onRestoreInstanceState(AbsListView.java:1807)
05-09 19:38:44.749: W/System.err(7188):     android.view.View.dispatchRestoreInstanceState(View.java:12806)
05-09 19:38:44.749: W/System.err(7188):     android.view.ViewGroup.dispatchThawSelfOnly(ViewGroup.java:2657)
05-09 19:38:44.749: W/System.err(7188):     android.widget.AdapterView.dispatchRestoreInstanceState(AdapterView.java:791)
05-09 19:38:44.749: W/System.err(7188):     android.view.ViewGroup.dispatchRestoreInstanceState(ViewGroup.java:2643)
05-09 19:38:44.749: W/System.err(7188):     android.view.View.restoreHierarchyState(View.java:12784)
05-09 19:38:44.749: W/System.err(7188):     com.tns.Runtime.callJSMethodNative(Native Method)
05-09 19:38:44.749: W/System.err(7188):     com.tns.Runtime.dispatchCallJSMethodNative(Runtime.java:861)
05-09 19:38:44.749: W/System.err(7188):     com.tns.Runtime.callJSMethodImpl(Runtime.java:726)
05-09 19:38:44.749: W/System.err(7188):     com.tns.Runtime.callJSMethod(Runtime.java:712)
05-09 19:38:44.749: W/System.err(7188):     com.tns.Runtime.callJSMethod(Runtime.java:693)
05-09 19:38:44.749: W/System.err(7188):     com.tns.Runtime.callJSMethod(Runtime.java:683)
05-09 19:38:44.749: W/System.err(7188):     com.tns.gen.android.support.v4.view.PagerAdapter_frnal_ts_helpers_l47_c38__PagerAdapterClassInner.instantiateItem(PagerAdapter_frnal_ts_helpers_l47_c38__PagerAdapterClassInner.java:24)
05-09 19:38:44.749: W/System.err(7188):     android.support.v4.view.ViewPager.addNewItem(ViewPager.java:943)
05-09 19:38:44.749: W/System.err(7188):     android.support.v4.view.ViewPager.populate(ViewPager.java:1157)
05-09 19:38:44.749: W/System.err(7188):     android.support.v4.view.ViewPager.setCurrentItemInternal(ViewPager.java:608)
05-09 19:38:44.749: W/System.err(7188):     android.support.v4.view.ViewPager.setCurrentItemInternal(ViewPager.java:570)
05-09 19:38:44.749: W/System.err(7188):     android.support.v4.view.ViewPager.setCurrentItem(ViewPager.java:551)
05-09 19:38:44.749: W/System.err(7188):     org.nativescript.widgets.TabLayout$TabClickListener.onClick(TabLayout.java:377)
05-09 19:38:44.749: W/System.err(7188):     android.view.View.performClick(View.java:4445)
05-09 19:38:44.749: W/System.err(7188):     android.view.View$PerformClick.run(View.java:18446)
05-09 19:38:44.749: W/System.err(7188):     android.os.Handler.handleCallback(Handler.java:733)
05-09 19:38:44.749: W/System.err(7188):     android.os.Handler.dispatchMessage(Handler.java:95)
05-09 19:38:44.749: W/System.err(7188):     android.os.Looper.loop(Looper.java:136)
05-09 19:38:44.749: W/System.err(7188):     android.app.ActivityThread.main(ActivityThread.java:5146)
05-09 19:38:44.749: W/System.err(7188):     java.lang.reflect.Method.invokeNative(Native Method)
05-09 19:38:44.749: W/System.err(7188):     java.lang.reflect.Method.invoke(Method.java:515)
05-09 19:38:44.749: W/System.err(7188):     com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:732)
05-09 19:38:44.749: W/System.err(7188):     com.android.internal.os.ZygoteInit.main(ZygoteInit.java:566)
05-09 19:38:44.749: W/System.err(7188):     dalvik.system.NativeStart.main(Native Method)
05-09 19:38:44.749: W/System.err(7188): File: "/data/data/info.rborn.asturplaya/files/app/tns_modules/ui/tab-view/tab-view.js, line: 61, column: 38
05-09 19:38:44.749: W/System.err(7188): StackTrace: 
05-09 19:38:44.749: W/System.err(7188): 	Frame: function:'PagerAdapterClassInner.instantiateItem', file:'/data/data/info.rborn.asturplaya/files/app/tns_modules/ui/tab-view/tab-view.js', line: 61, column: 39
05-09 19:38:44.749: W/System.err(7188): 	at com.tns.Runtime.callJSMethodNative(Native Method)
05-09 19:38:44.749: W/System.err(7188): 	at com.tns.Runtime.dispatchCallJSMethodNative(Runtime.java:861)
05-09 19:38:44.749: W/System.err(7188): 	at com.tns.Runtime.callJSMethodImpl(Runtime.java:726)
05-09 19:38:44.749: W/System.err(7188): 	at com.tns.Runtime.callJSMethod(Runtime.java:712)
05-09 19:38:44.749: W/System.err(7188): 	at com.tns.Runtime.callJSMethod(Runtime.java:693)
05-09 19:38:44.749: W/System.err(7188): 	at com.tns.Runtime.callJSMethod(Runtime.java:683)
05-09 19:38:44.749: W/System.err(7188): 	at com.tns.gen.android.support.v4.view.PagerAdapter_frnal_ts_helpers_l47_c38__PagerAdapterClassInner.instantiateItem(PagerAdapter_frnal_ts_helpers_l47_c38__PagerAdapterClassInner.java:24)
05-09 19:38:44.749: W/System.err(7188): 	at android.support.v4.view.ViewPager.addNewItem(ViewPager.java:943)
05-09 19:38:44.749: W/System.err(7188): 	at android.support.v4.view.ViewPager.populate(ViewPager.java:1157)
05-09 19:38:44.749: W/System.err(7188): 	at android.support.v4.view.ViewPager.setCurrentItemInternal(ViewPager.java:608)
05-09 19:38:44.749: W/System.err(7188): 	at android.support.v4.view.ViewPager.setCurrentItemInternal(ViewPager.java:570)
05-09 19:38:44.749: W/System.err(7188): 	at android.support.v4.view.ViewPager.setCurrentItem(ViewPager.java:551)
05-09 19:38:44.749: W/System.err(7188): 	at org.nativescript.widgets.TabLayout$TabClickListener.onClick(TabLayout.java:377)
05-09 19:38:44.749: W/System.err(7188): 	at android.view.View.performClick(View.java:4445)
05-09 19:38:44.749: W/System.err(7188): 	at android.view.View$PerformClick.run(View.java:18446)
05-09 19:38:44.749: W/System.err(7188): 	at android.os.Handler.handleCallback(Handler.java:733)
05-09 19:38:44.759: W/System.err(7188): 	at android.os.Handler.dispatchMessage(Handler.java:95)
05-09 19:38:44.759: W/System.err(7188): 	at android.os.Looper.loop(Looper.java:136)
05-09 19:38:44.759: W/System.err(7188): 	at android.app.ActivityThread.main(ActivityThread.java:5146)
05-09 19:38:44.759: W/System.err(7188): 	at java.lang.reflect.Method.invokeNative(Native Method)
05-09 19:38:44.759: W/System.err(7188): 	at java.lang.reflect.Method.invoke(Method.java:515)
05-09 19:38:44.759: W/System.err(7188): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:732)
05-09 19:38:44.759: W/System.err(7188): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:566)
05-09 19:38:44.759: W/System.err(7188): 	at dalvik.system.NativeStart.main(Native Method)
05-09 19:38:44.759: W/System.err(7188): Caused by: java.lang.ClassCastException: android.view.AbsSavedState$1 cannot be cast to android.widget.AbsListView$SavedState
05-09 19:38:44.759: W/System.err(7188): 	at android.widget.AbsListView.onRestoreInstanceState(AbsListView.java:1807)
05-09 19:38:44.759: W/System.err(7188): 	at android.view.View.dispatchRestoreInstanceState(View.java:12806)
05-09 19:38:44.759: W/System.err(7188): 	at android.view.ViewGroup.dispatchThawSelfOnly(ViewGroup.java:2657)
05-09 19:38:44.759: W/System.err(7188): 	at android.widget.AdapterView.dispatchRestoreInstanceState(AdapterView.java:791)
05-09 19:38:44.759: W/System.err(7188): 	at android.view.ViewGroup.dispatchRestoreInstanceState(ViewGroup.java:2643)
05-09 19:38:44.759: W/System.err(7188): 	at android.view.View.restoreHierarchyState(View.java:12784)
05-09 19:38:44.769: W/System.err(7188): 	... 24 more

~~~

**Comment by tsonevn** [comment]
Hi @rborn, 
Thank you for reporting this issue,
I made sample app. Unfortunately I confirm that this is an real problem while switching between the tabs.
I am attaching my sample project.

https://github.com/tsonevn/TabViewNativeScriptGoogleMaps.git

Regards,
@tsonevn

**Comment by rborn** [comment]
Any news on this? Thank you 😄

**Comment by rborn** [comment]
Spoke too soon, I believe #2133 fixes it 😄 
I'll test more and let you know
Thank you

**Comment by tsonevn** [comment]
fixed  #2133
